react/features/app/actions.js

```diff
@@ -31,176 +31,72 @@ declare var APP: Object;
-    // default.
-    if (!location || !location.host) {
-        const defaultLocation = parseURIString(getDefaultURL(getState));
-
-        if (location) {
-            location.host = defaultLocation.host;
-
-            // FIXME Turn location's host, hostname, and port properties into
-            // setters in order to reduce the risks of inconsistent state.
-            location.hostname = defaultLocation.hostname;
-            location.pathname
-                = defaultLocation.pathname + location.pathname.substr(1);
-            location.port = defaultLocation.port;
-            location.protocol = defaultLocation.protocol;
-        } else {
-            // eslint-disable-next-line no-param-reassign
-            location = defaultLocation;
+    return async(dispatch: Dispatch<any>, getState: Function) => {
+        let location = parseURIString(uri);
+
+        // If the specified location (URI) does not identify a host, use the app's
+        // default.
+        if (!location || !location.host) {
+            const defaultLocation = parseURIString(getDefaultURL(getState));
+
+            if (location) {
+                location.host = defaultLocation.host;
+
+                // FIXME Turn location's host, hostname, and port properties into
+                // setters in order to reduce the risks of inconsistent state.
+                location.hostname = defaultLocation.hostname;
+                location.pathname
+                    = defaultLocation.pathname + location.pathname.substr(1);
+                location.port = defaultLocation.port;
+                location.protocol = defaultLocation.protocol;
+            } else {
+                // eslint-disable-next-line no-param-reassign
+                location = defaultLocation;
+            }
         }
-    }
 
-    location.protocol || (location.protocol = 'https:');
+        location.protocol || (location.protocol = 'https:');
+        const locationURL = new URL(location.toString());
 
-    return _appNavigateToMandatoryLocation(dispatch, getState, location);
-}
+        dispatch(configWillLoad(locationURL));
 
-/**
- * Loads config.js from a specific host.
- *
- * @param {Dispatch} dispatch - The redux {@code dispatch} function.
- * @param {Function} getState - The redux {@code getState} function.
- * @param {Object} location - The location URI which specifies the host to load
- * the config.js from.
- * @private
- * @returns {Promise<Object>}
- */
-function _loadConfig(
-        dispatch: Dispatch<any>,
-        getState: Function,
-        { contextRoot, host, protocol, room }) {
-    // XXX As the mobile/React Native app does not employ config on the
-    // WelcomePage, do not download config.js from the deployment when
-    // navigating to the WelcomePage - the perceived/visible navigation will be
-    // faster.
-    if (!room && typeof APP === 'undefined') {
-        return Promise.resolve();
-    }
```

Comment:
This no-oping for mobile has been removed? So now mobile loads config, yeah?

Reply:
Yes, we have plans with it for later